tests/integration/helpers.py

```diff
@@ -101,6 +104,7 @@ async def build_and_deploy(
             status=status,
             raise_on_blocked=True,
             timeout=1000,
+            check_freq=3,
```

Comment:
I think we should set `raise_on_error=False` here instead of changing the frequency everywhere. If the volume error is transient and happens only on deploy, we should only try to suppress it on deploy.

Reply:
But it doesn't happen on deploy exactly, it happens on wait_for_idle (and we probably don't want to disable raising on error completely on there)

Reply:
It is on an early stage on the charm deployment (during agent install) but it's the wait_on_idle call who raises the error

Reply:
`build_and_deploy()` will deploy and `wait_for_idle()`. If the error is happening on the wait after the deploy, that's where we should guard for the error. The worse case scenario for setting `raise_on_error=False` only during `build_and_deploy()` is that we will have to wait the full timeout for the deploy to fail, since if the error doesn't heal, `wait_for_idle()` should still fail in the end. Tweaking `check_freq` everywhere, on the other hand, could also hide other issues beyond the volume mount error.

Reply:
I'm not sure I get your point: the way I see it, increasing check frequency would only "hide" errors which are transient and resolve in < 3s, which to me is fine. Avoiding raise on errors, on the other hand, will not only hide the same transient errors, but will hide non-transient errors as well, and we'll only see what happened after timeout. I personally don't see the benefit.

Reply:
Testing Try-catch with retry approach here https://github.com/canonical/postgresql-k8s-operator/pull/557

Reply:
Try-catching the error didn´t work so well, since it requires some effort to effectively retry the deployment on `build_and_deploy` helper, and still in a lot tests we don't even use this function to deploy (some tests where we deploy and relate 2 or more charms at the same time).

After some test and consideration, I think Dragomir's suggestion is indeed the best approach here: Adding `raise_on_error=False` to `build_and_deploy`'s `wait_for_idle' call (and, in the case of tests who don´t use it, to the first `wait_for_idle` call right after deploy) is a sure way to avoid the bug in the specific place where it happens + If anything goes wrong later in the execution, by then we'll have logs to help debug the problem.

I'll adapt this PR to remove all the `check_freq`s and use `raise_on_error=False` in the places where I think it makes sense. LMK what you think.

Reply:
Huh, it ended up a bit ugly since a lot of places either don't use `build_and_deploy`, or use it with `wait_for_idle=False` and waiting afterwards... LMK if you agree with the changes or see a better way.